LoopKit/DeviceManager/CGMManager.swift

```diff
@@ -86,14 +88,30 @@ public protocol CGMManager: DeviceManager {
     /// The representation of the device for use in HealthKit
     var device: HKDevice? { get }
 
-    /// The current status of the cgm
-    var cgmStatus: CGMManagerStatus { get }
+    /// The current status of the cgm manager
+    var cgmManagerStatus: CGMManagerStatus { get }
 
     /// Performs a manual fetch of glucose data from the device, if necessary
     ///
     /// - Parameters:
     ///   - completion: A closure called when operation has completed
     func fetchNewDataIfNeeded(_ completion: @escaping (CGMReadingResult) -> Void) -> Void
+
+    /// Adds an observer of changes in CGMManagerStatus
+    ///
+    /// Observers are held by weak reference.
+    ///
+    /// - Parameters:
+    ///   - observer: The observing object
+    ///   - queue: The queue on which the observer methods should be called
+    func addStatusObserver(_ observer: CGMManagerStatusObserver, queue: DispatchQueue)
```

Comment:
The use of observer/notifications pattern seems to be increasing. Should we consider instead just making `CGMManagerStatus` a subclass of `ObservableObject`?

cc: @ps2

Reply:
Might be worth exploration. I'm not very familiar using this outside of SwiftUI>

Reply:
@darinkrauss If we use this pattern, won't `CGMManager` be the `ObserverableObject` and `CGMManagerStatus` be the `@Published` property, and would mean any that wish to observe need to retain a reference to the `CGMManager`.

Reply:
Or do you mean change CGMManagerStatus to a `class`?

Reply:
Ah, yes, I see what you mean. I was thinking that the CGMManagerClass was fixed and just the properties changed within it. So, yes, `CGMManager` would have to be the `ObservableObject` and I don't think we are ready to do that. So, carry on. :)